zokrates_core/src/proof_system/bellman/groth16.rs

```diff
@@ -203,6 +221,23 @@ impl<T: Field> ProofSystem<T> for G16 {
     }
 }
 
+mod serialization {
+    use proof_system::{G1Affine, G2Affine};
+    use zokrates_field::Field;
+    use pairing::{CurveAffine, Engine, from_hex};
+
+    pub fn to_g1<E: Engine>(g1: G1Affine) -> E::G1Affine {
```

Comment:
For consistency, how about having both `to_g1` and `to_g2` both be generic over `T: Field`? `<T::BellmanEngine as Engine>` can be used to get the associated bellman engine